### ISSUE TITLE
Allow using IxCategoryFilter in a form

### ISSUE THREAD

**Issue by fakhir** [siemens/ix#755]
### Suggestion / feature request

I am creating a form in which I need to use an element for editing tags. I decided to use IxCategoryFilter for this purpose within my form (in conjunction with react-hook-form). But the problem is that the moment I select a suggested tag within the IxCategoryFilter, the form immediately gets submitted. I would expect the form to remain unsubmitted and only the IxCategoryFilter should show the selected tag and wait for the user to click the form submit button.

I have also created a sandbox here to demonstrate this problem (selecting a tag would cause a page reload due to premature form submission):

https://codesandbox.io/s/sharp-sinoussi-x2q2rz

I'm not sure whether this is a bug report or a feature request. Kindly advise how to address this situation.

**Comment by fakhir** [comment]
After submitting this request, I realized that using IxSelect is probably the recommended element in this case, instead of IxCategoryFilter. And this IxCategoryFilter isn't meant to be used within a form? Would there be any use-cases where using a category filter would be needed within a form?

**Comment by robert-wettstaedt** [comment]
Maybe related? #691

**Comment by danielleroux** [comment]
TLDR: I will close the issue here, because category-filter will not be an form element because. The value of an HTML form input element can only be a string.

This means it is not possible to directly set a complex JavaScript object (like an array, object, or function) as the value for a form input, nor will such objects be included properly in the form submission. If you try to assign a non-string value (such as a custom object) to input.value via JavaScript, it will automatically be converted to its string representation (such as "[object Object]").
